gpii/node_modules/packagekit/nodepackagekit/nodepackagekit.cc

```diff
@@ -30,12 +38,10 @@ static gboolean DEBUG = FALSE;
 static gchar TAG[] = "[nodePackagekit]";
 
 static std::basic_string<gchar>
-utf8StringFromValue(v8::Handle<v8::Value> value)
+utf8StringFromValue(Handle<Value> value)
 {
-    v8::Local<v8::String> string = value->ToString();
-    gchar *buffer = g_new(gchar, string->Utf8Length());
-    string->WriteUtf8(buffer);
-    return buffer;
+    Local<String> string = value.As<String>();
+    return (*String::Utf8Value(string));
```

Comment:
Hard to tell whether we have lost a memory leak or gained one here :)
Perhaps someone can walk me through the expectation here - we believe that the Utf8Value object will be destructed on the stack, the "string" will be destructed on the stack shortly afterwards, deallocating its buffer, which has been copied into the returned string in the meantime?
http://stackoverflow.com/a/1032671/5041298 reminds us that every construction of a std::string will do a heap allocation as it copies the buffer. This suggests that we once had a memory leak - can we be clear that we don't have one any more?
It looks like there are some Nan utilities we should use to make this clearer and more portable rather than relying on V8 primitives, even if we succeed in convincing ourselves that this code is valid.